oarepo_requests/resolvers/ui.py

```diff
@@ -0,0 +1,102 @@
+from invenio_records_resources.resources.errors import PermissionDeniedError
+from invenio_users_resources.proxies import current_users_service
+
+from oarepo_requests.utils import get_matching_service_for_refdict
+
+
+def fallback_result(reference):
+    id = list(reference.values())[0]
+    return {"reference": reference, "type": "user", "label": f"id: {id}"}
+
+
+def user_entity_reference_ui_resolver(identity, data):
+    reference = data["reference"]
+    user_id = reference["user"]
+    try:
+        user_search = current_users_service.read(identity, user_id)
+    except PermissionDeniedError:
+        return fallback_result(reference)
+    if user_search.data["username"] is None:  # username undefined?
+        if "email" in user_search.data:
+            label = user_search.data["email"]
+        else:
+            label = f"id: {user_search.data['id']}"
+    else:
+        label = user_search.data["username"]
+    ret = {
+        "reference": reference,
```

Comment:
is the "reference" needed?

Reply:
according to specification here https://linear.app/ducesnet/issue/FE-116/display-requests-on-record-page